checkov/common/runners/base_runner.py

```diff
@@ -125,16 +126,19 @@ def load_external_checks(self, external_checks_dir: List[str]) -> None:
     def get_graph_checks_report(self, root_folder: str, runner_filter: RunnerFilter) -> Report:
         return Report(check_type="not_defined")
 
-    def run_graph_checks_results(self, runner_filter: RunnerFilter, report_type: str) -> dict[BaseGraphCheck, list[_CheckResult]]:
+    def run_graph_checks_results(self, runner_filter: RunnerFilter, report_type: str, graph: Graph | None = None
```

Comment:
why do you need to send the `graph` here as well? (instead of always using `get_reader_endpoint`)

Reply:
We want to run on a specific graph, because we have multiple graphs now, not one as before

Reply:
sound right, hadn't though about it.